src/module/helpers/migrations.ts

```diff
@@ -125,13 +127,44 @@ async function normalizeDelveTableRows() {
 	})
 }
 
+/**
+ * Migration 5: Convert challenge ranks from strings to numbers.
+ */
+async function convertRanksToNumbers() {
+	function toNumberRank(oldRank: unknown): null | undefined | ChallengeRank {
+		// nullish value - return unchanged
+		if (oldRank == null) return oldRank
+
+		switch (typeof oldRank) {
+			case 'string':
+				return (
+					ChallengeRank[capitalize(oldRank) as keyof typeof ChallengeRank] ??
+					ChallengeRank.Troublesome
+				)
+			case 'number':
+				// already a number - no change required
+				return oldRank
+			default:
+				// something weird -- set a fallback
+				return ChallengeRank.Troublesome
+		}
+	}
+	async function updateDoc(doc: any) {
+		if (doc.system?.rank != null)
```

Comment:
Oh that's weird. In JS `null == undefined` is `true`. 🤦🏼

Reply:
yeah, that formulation is apparently what standardjs wants us to use instead of `if (!doc.system?.rank)`. a little annoying, but if it's safer and the linter can automatically fix it, might as well :shrug: